src/main/java/com/stablekernel/rxespresso/RxIdlingResource.java

```diff
@@ -129,12 +133,37 @@ public void onNext(T t) {
 
     private <T> void onFinally(Observable.OnSubscribe<T> onSubscribe, final String finalizeCaller) {
         int activeSubscriptionCount = subscriptions.decrementAndGet();
+        removeOnSubscribe(onSubscribe);
+
         if (LOG_LEVEL >= DEBUG) {
-            Log.d(TAG, onSubscribe + " - " + finalizeCaller + ": " + activeSubscriptionCount);
+            Log.d(TAG, "Subscription finished because: " + finalizeCaller + ". Active subscription count is now: " + activeSubscriptionCount);
+
+            if (LOG_LEVEL >= VERBOSE) {
+                // remaining subscriptions:
+                Log.d(TAG, ">>>>>>> remaining subscriptions ");
+                printStackTraces();
+                Log.d(TAG, "<<<<<<<<");
+            }
```

Comment:
This might be better inside `isIdleNow()` when `isIdle == false`.  That way it doesn't log the stack traces for every subscription ending but when Espresso is asking if it's idle.